src/main/java/at/hannibal2/skyhanni/features/mining/FlowstateHelper.kt

```diff
@@ -0,0 +1,168 @@
+        "mining.pickobulus.blockdestroy",
+        "§7Your §r§aPickobulus §r§7destroyed §r§e(?<amount>\\d+) §r§7blocks!"
+    )
+
+    enum class FlowstateElements(val label: String, var renderable: Renderable) {
+        TITLE("§d§lFlowstate Helper", Renderable.string("§d§lFlowstate Helper")),
+        TIMER("Time Remaining: §b6.71", Renderable.string("")),
+        STREAK("Streak: §7234", Renderable.string("")),
+        SPEED("§6+600⸕", Renderable.string("")),
+        ;
+
+        override fun toString() = label
+
+        companion object {
+            @JvmField
+            val defaultOption = listOf(
+                TITLE, TIMER, STREAK, SPEED
+            )
+        }
+    }
+
+    @HandleEvent(onlyOnSkyblock = true)
+    fun onBlockMined(event: OreMinedEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (flowstateCache == null) return
+
+        streakEndTimer = SimpleTimeMark.now().plus(10.seconds)
+        blockBreakStreak += event.extraBlocks.values.sum()
+        createDisplay()
+        displayDirty = true
+        attemptClearDisplay()
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) { //TODO: Remove once #2540 is merged
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (flowstateCache == null) return
+
+        pickobulusPattern.matchMatcher(event.message) {
+            streakEndTimer = SimpleTimeMark.now().plus(10.seconds)
+            blockBreakStreak += group("amount").toInt()
+            createDisplay()
+            attemptClearDisplay()
+            displayDirty = true
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+
+        attemptClearDisplay()
+    }
+
+    private fun attemptClearDisplay() {
+        if (streakEndTimer.isInFuture()) return
+        blockBreakStreak = 0
+        displayDirty = true
+        createDisplay()
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (!config.flowstateHelper) return
+        if (flowstateCache == null) return
+
+        if (display == null || streakEndTimer.isInFuture()) createDisplay()
+        display?.let {
+            config.flowstateHelperPosition.renderRenderables(
+                it,
+                1,
+                "Flowstate Helper",
+                true
+            )
+        }
+    }
+
+    private fun createDisplay() {
+        if (displayDirty) {
+            displayDirty = false
+            createDisplayTimer()
+            createDisplayBlock()
+            createDisplaySpeed()
+        } else {
+            createDisplayTimer()
+        }
+        display = config.flowstateHelperAppearance.map { it.renderable }
+    }
+
+    private fun createDisplayTimer(): Renderable {
+        var timeRemaining = streakEndTimer.minus(SimpleTimeMark.now()).inPartialSeconds
```

Comment:
work with duration, not double. compare the time below with `< 0.seconds`. format the time for the rendererable with `timeRemaining.format()`. Optionaly define when millis should be shown or not.